Ladder step3

undervineg opened the pull request:
- 이름 입력, 5자 제한, 이름 출력
- 플레이어 및 게임 구조체 활용
까지 완료하였습니다.
enum, class 등도 말씀하셨는데, 지금은 struct만 활용한 상태입니다.

커밋 메시지는 좀 더 자세하게 쓰려고 노력중입니다. 작성 규칙은 차차 개선해 나가겠습니다.
검토 부탁드립니다.

undervine committed:
README.md file updated
스위프트 문법 - 타입메소드, 열거형, 스위치, 확장기능 설명 추가

undervine committed:
플레이어 구조체 파일 추가.
기본 로직은 게임 구조체로 옮김.
플레이어 이름 5자 이상은 무시, 마지막 줄에 이름 출력.

godrm reviewed (changes requested):
LadderGame 내부에 있는 입력 받는 기능(InputView)과 출력하는 부분(OutputView)도 분리해보세요. 
커밋 단위도 작업 단위로 쪼갤 수 있는지 고민해보세요. 그래야 더 구체적이고 정확한 내용을 표현할 수 있습니다. 나중에 본인이 어떤 내용을 어디서 수정했는지 확인한다고 상상해보세요.
위 사항을 개선하고 다시 커밋하세요.

undervine committed:
Devide InputView and ResultView

Devide Main into InputView and ResultView.

undervine committed:
Add 커밋 메시지 작성 규칙 to README.md file

undervine committed:
Add 요구사항: 객체역할분담 내용 in README.md file

godrm reviewed (commented):
객체 분리가 잘 된 편입니다.
LadderGame와 InputView, OutputView 관계를 좀 더 개선해보세요. Input/Output은 다른 객체로 대체가 가능하도록 LadderGame과 데이터를 주고 받는게 좋습니다.
Main에서는 while() 같은 로직이 들어가지 않고 LadderGame과 InputView, OutputView 객체 인스턴스만 만들어서 연결해주는 역할을 하도록 개선하세요.

undervine committed:
Improve independency of InputView and OutputView

Improve independecy of InputView and OutputView to make it reusable.

undervine committed:
Make main() function

Make main() function to remove while() statement.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -0,0 +1,114 @@
+    func makeTransverseLineBetween(_ row: Int, and col: Int)->[[Bool]]{
+        // 사다리가 그려질 세로(column) 공간의 개수는 사람수보다 1개 작음.(사이공간이므로)
+        let maxNumberOfCol = col - 1
+        // 여러 개의 사다리(-) 위치를 담을 배열. 원소는 false로 채운다. 사다리가 있는 경우 true로 바꿀 예정.
```
이전 코드는 커밋 이력이나 기존 브랜치에 모두 있으니까 사용하지 않는 코드는 과감하게 지우세요.

undervineg replied:
네 알겠습니다 ㅎㅎ 안 지운 게 있었네요

undervine committed:
Remove previous code comments

Remove makeTransverseLienBetween() function treated as comments previously.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -0,0 +1,114 @@
+
+struct LadderGame {
+    var heights: Int
+    var names: [LadderPlayer]
```
var heights 와 var names를 접근 관리를 위해서 private 이면서 get 이 가능하도록 개선해보세요.
객체 속성을 객체 외부에서 바꿀수 있도록 감추도록 작성하세요.

undervineg replied:
네 알겠습니다.

godrm commented on LadderGame/LadderGame/LadderGame.swift:
```diff
@@ -0,0 +1,114 @@
+    
+    // 전체 사다리(-) 위치 반환. 사다리가 있는 경우, 사다리 위치를 가리키는 인덱스에 true값 넣어 배열 반환.
+    var transverLine: [[Bool]] {
+        // 사다리가 그려질 세로(column) 공간의 개수는 사람수보다 1개 작음.(사이공간이므로)
```
이 속성은 이미 만들어진 2차원 사다리 배열을 전달하는건지,
호출할때마다 새로운 사다리를 만들어서 배열을 전달하는건지 역할을 하나만 하도록 명확하게 분리해보세요.
매번 다른 배열이 넘어온다면 메소드로 만드는게 더 좋을 것 같습니다. 
속성에만 접근했는데 값이 바뀌는 것은 그 객체를 사용할 때 실수할 가능성이 많아집니다.

undervineg replied:
네 알겠습니다.

undervine committed:
Make properties in struct private

Make var heights, var names in struct LadderGame private not to be changed from outside of the file.

undervine committed:
Change computable property to function

Change computable var 'transverLine' to function because it's expendable generating [[Bool]] array every time being accessed.

godrm commented:
피드백 반영이 잘 되었고, 객체 역할 분담도 적당합니다.
머지하도록 하겠습니다. 앞으로 다음 미션을 풀더라도 계속해서 피드백 받았던 사항을 유지하도록 해보세요.
